test/onnx/test_pytorch_onnx_onnxruntime.py

```diff
@@ -5134,6 +5139,18 @@ def forward(self, x, y, z):
         z = torch.randn(2, 4, 5)
         self.run_test(ConcatModel(), (x, y, z))
 
+    @skipIfUnsupportedMinOpsetVersion(12)
+    def test_einsum_input_type_promotion(self):
+        class EinsumModel(torch.nn.Module):
+            def forward(self, fp16, fp32):
+                return torch.einsum("bkhd,bqhd->bqkh", fp16, fp32)
```

Comment:
The exported ONNX model output is fp32, while pytorch output is bfloat16. I'd recommend updating `ort_compare_with_pytorch` to also check if dtype matches.

Do we know how `torch.autocast` works underneath? It does not insert cast into the traced graph, but somehow end up getting inputs casted.

Reply:
I didnt look into much detail, but autocast is a kind of context manager to store the actual dtype used during computation while preserving the original inputs dtype in the aten graph